internal/privi/privi.go

```diff
@@ -79,3 +79,29 @@ func (p *store) getRecord(
 
 	return p.repo.getRecord(string(targetDID), rkey)
 }
+
+// Same signature as repo.getBlob, wrapped with permissions
+func (p *store) getBlob(requestorDID string, ownerDID string, cid string) (string /* mimetype */, []byte /* raw blob */, error) {
+	refs, err := p.repo.getBlobRefs(cid, ownerDID)
+	if err != nil {
+		return "", nil, err
+	}
+
+	ok := false
+	for _, ref := range refs {
```

Comment:
this seems good for now but we would be able to do this in a single query if we merged the permission store and repo. I don't think we're getting much from keeping them independent